src/variety/kuroclone.js

```diff
@@ -0,0 +1,232 @@
+		},
+		mouseinput_auto: function() {
+			if (this.puzzle.playmode) {
+				if (this.mousestart || this.mousemove) {
+					this.inputcell();
+				}
+			} else if (this.puzzle.editmode) {
+				if (this.mousestart || this.mousemove) {
+					if (this.isBorderMode()) {
+						this.inputborder();
+					} else {
+						this.inputdirec();
+					}
+				} else if (this.mouseend && this.notInputted()) {
+					this.inputqnum();
+				}
+			}
+		}
+	},
+
+	//---------------------------------------------------------
+	// キーボード入力系
+	KeyEvent: {
+		enablemake: true,
+		moveTarget: function(ca) {
+			if (ca.match(/shift/)) {
+				return false;
+			}
+			return this.moveTCell(ca);
+		},
+
+		keyinput: function(ca) {
+			if (this.key_inputdirec(ca)) {
+				return;
+			}
+			this.key_inputqnum(ca);
+		}
+	},
+
+	//---------------------------------------------------------
+	// 盤面管理系
+	Cell: {
+		numberRemainsUnshaded: true,
+		minmum: 1,
+		maxnum: 99
+	},
+	Board: {
+		hasborder: 1
+	},
+	BoardExec: {
+		adjustBoardData: function(key, d) {
+			this.adjustNumberArrow(key, d);
+		}
+	},
+	AreaRoomGraph: {
+		enabled: true,
+		hastop: false
+	},
+	AreaShadeGraph: {
+		enabled: true
+	},
+
+	//---------------------------------------------------------
+	// 画像表示系
+	Graphic: {
+		gridcolor_type: "LIGHT",
+
+		enablebcolor: true,
+		bgcellcolor_func: "qsub1",
+
+		paint: function() {
+			this.drawBGCells();
+			this.drawGrid();
+			this.drawShadedCells();
+			this.drawArrowNumbers();
+			this.drawBorders();
+			this.drawChassis();
+			this.drawBoxBorders(false);
+			this.drawTarget();
+		}
+	},
+
+	//---------------------------------------------------------
+	// URLエンコード/デコード処理
+	Encode: {
+		decodePzpr: function(type) {
+			this.decodeBorder();
+			this.decodeArrowNumber16();
+		},
+		encodePzpr: function(type) {
+			this.encodeBorder();
+			this.encodeArrowNumber16();
+		}
+	},
+	//---------------------------------------------------------
+	FileIO: {
+		decodeData: function() {
+			this.decodeAreaRoom();
+			this.decodeCellDirecQnum();
+			this.decodeCellAns();
+		},
+		encodeData: function() {
+			this.encodeAreaRoom();
+			this.encodeCellDirecQnum();
+			this.encodeCellAns();
+		}
+	},
+
+	//---------------------------------------------------------
+	// 正解判定処理実行部
+	AnsCheck: {
+		checklist: [
+			"checkSideAreaShadeCell",
+			"checkArrowNumber",
+			"checkUnitsCount",
+			"checkUnitsShape"
+		],
+
+		// From shimaguni.js
+		checkSideAreaShadeCell: function() {
+			this.checkSideAreaCell(
+				function(cell1, cell2) {
+					return cell1.isShade() && cell2.isShade();
+				},
+				false,
+				"cbShade"
+			);
+		},
+
+		// Check unit size pointed by arrow
+		checkArrowNumber: function() {
+			var bd = this.board;
+			for (var c = 0; c < bd.cell.length; c++) {
+				var cell = bd.cell[c];
+				if (!cell.isValidNum() || cell.qdir === 0) {
+					continue;
+				}
+				var pos = cell.getaddr();
+				var cell2 = pos.movedir(cell.qdir, 2).getc();
+
+				if (cell2.sblk) {
```

Comment:
I think I prefer a separate check function for "The cell next to the arrow is not shaded."